controllers/helper.go

```diff
@@ -302,17 +365,7 @@ func syncDaemonSet(ctx context.Context, client k8sclient.Client, scheme *runtime
 
 		if equality.Semantic.DeepEqual(in.OwnerReferences, ds.OwnerReferences) &&
 			equality.Semantic.DeepDerivative(in.Spec, ds.Spec) {
-			// DeepDerivative has issue detecting nodeAffinity change
```

Comment:
Why is this part not needed anymore?

Reply:
because `equality.Semantic.DeepDerivative` is able to handle that